src/main/java/cz/cvut/kbss/termit/service/business/SparqlExternalVocabularyService.java

```diff
@@ -0,0 +1,236 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+package cz.cvut.kbss.termit.service.business;
+
+import cz.cvut.kbss.jopa.model.MultilingualString;
+import cz.cvut.kbss.termit.model.RdfsResource;
+import cz.cvut.kbss.termit.event.VocabularyCreatedEvent;
+import cz.cvut.kbss.termit.model.Vocabulary;
+import cz.cvut.kbss.termit.model.acl.AccessControlList;
+import cz.cvut.kbss.termit.service.repository.VocabularyRepositoryService;
+import cz.cvut.kbss.termit.util.Configuration;
+import cz.cvut.kbss.termit.util.Utils;
+import jakarta.annotation.Nonnull;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.net.URI;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import org.eclipse.rdf4j.query.BindingSet;
+import org.eclipse.rdf4j.query.GraphQuery;
+import org.eclipse.rdf4j.query.GraphQueryResult;
+import org.eclipse.rdf4j.query.QueryEvaluationException;
+import org.eclipse.rdf4j.query.TupleQuery;
+import org.eclipse.rdf4j.query.TupleQueryResult;
+import org.eclipse.rdf4j.repository.RepositoryConnection;
+import org.eclipse.rdf4j.repository.RepositoryException;
+import org.eclipse.rdf4j.repository.sparql.SPARQLRepository;
+import org.eclipse.rdf4j.rio.RDFFormat;
+import org.eclipse.rdf4j.rio.Rio;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * Business logic concerning external vocabularies imported from SPARQL repository.
+ */
+@Service
+public class SparqlExternalVocabularyService implements ExternalVocabularyService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SparqlExternalVocabularyService.class);
+
+    private final Configuration config;
+    private final VocabularyRepositoryService repositoryService;
+
+    private final AccessControlListService aclService;
+    private ApplicationEventPublisher eventPublisher;
+
+    private static final String LIST_AVAILABLE_VOCABULARIES_QUERY = "import/listAvailableVocabularies.rq";
+    private static final String EXPORT_FULL_VOCABULARY_QUERY = "import/exportFullVocabulary.rq";
+
+    public SparqlExternalVocabularyService(VocabularyRepositoryService repositoryService,
+                                           AccessControlListService aclService,
+                                           Configuration config) {
+        this.repositoryService = repositoryService;
+        this.aclService = aclService;
+        this.config = config;
+    }
+    
+/**
+ * Sends a SPARQL query to fetch list of available vocabularies.
```

Comment:
Still broken formatting - the comment is not correctly indented. Moreover, the comment is inaccurate (the method does not return null on failure) and can probably be removed altogether, because the method declaration in `ExternalVocabularyService` already has javadoc.